File: sources/api/Models/Payement/Payment.cs
using DotNetAPI.Models.Debt;
using DotNetAPI.Models.User;

namespace DotNetAPI.Models.Payment
{
    public class Payment
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User.User User { get; set; }
        public int GroupId { get; set; }
        public Group.Group Group { get; set; }
        public float Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public int? DebtAdjustmentId { get; set; }
        public DebtAdjustment? DebtAdjustment { get; set; }
        public int type { get; set; }
    }
}

File: sources/mobile/app/src/main/java/com/console/ratcord/domain/entity/payment/Payment.kt
package com.console.ratcord.domain.entity.payment

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.console.ratcord.domain.entity.user.UserExtraMinimal
import kotlinx.serialization.Serializable

@Serializable
@Entity(tableName = "Payment")
data class Payment(
    @PrimaryKey val id: Int,
    val user: UserExtraMinimal,
    val userInCredit: UserExtraMinimal,
    val groupId: Int,
    val amount: Float,
    val debtAdjustmentId: Int?,
    val paymentDate: String,
    val type: String,
    val image: String
)